README.md

```diff
@@ -61,6 +63,77 @@ You will find the API is very similar to the Java/iOS versions except that:
    classMember = new Firebase("mypath");  //holds a reference until the behavior is released
    classMember.ValueChanged += (...) => {...} ;
    ```
+
+   But beware, if you are going to access child objects you need to store these in variables too, otherwise GC will delete the reference as well. So instead of this:
+  ```C#
+   classMember = new Firebase("mypath");  //holds a reference until the behavior is released
+   classMember.Child("somechild").ValueChanged += (...) => {...} ;
+   ```
+
+   do this
+
+   ```C#
+   IFirebase fb;
+   IFirebase sampleChild;
+   void someFunction() {
+       fb = new Firebase("mypath");  //holds a reference until the behavior is released
+       sampleChild = fb.Child("somechild"); // holds another child reference so GC doens't remove the reference to the child
+       sampleChild.ValueChanged += (...) => {...} ;
+   }
+   ```
+
+ If you are referencing objects correctly, you will likely need to re-open your Unity project because your event response function contained an error. At this moment an error in the event response function causes Firebase Unity to stop working during your entire Unity Editor session. You can debug this by catching the exception on the highest level, re-opening your Unity project, and running your scene again. You can now see the error and fix it. Note that you will need to re-open your project on every attempt until we fix this bug. After you have resolved your exception, your firebase event should be received normally again. 
+
+ For example: 
+   ```C#
+   IFirebase fb;
+   IFirebase sampleChild;
+   void someFunction() {
+       try {
+           fb = new Firebase("mypath");  //holds a reference until the behavior is released
+           sampleChild = fb.Child("somechild"); // holds another child reference so GC doens't remove the reference to the child
+           sampleChild.ValueChanged += (...) => {...} ;
+           someBadFunction();
+      } catch(Exception e) {
+        Debug.Log("Firebase Event Exception:");
+        Debug.Log(e);
+      }
+   }
+   ```
+
+ * Threading: Unity does not allow game object modifications from any thread other than the Main thread. Firebase operates on a separate thread for performance reasons, and so you can not directly edit game objects from your firebase event responses. You can solve this by adding a Queue with actions to be fulfilled on the main thread. Create a new script inside Unity, add it to the game object you just added to the scene. Then insert this code.
+
+  ```C#
+  using UnityEngine;
+  using System.Collections;
+  using System.Collections.Generic;
+  using System;
+
+  public class ExampleMainThreadQueue : MonoBehaviour {
+
+    public readonly static Queue<Action> ExecuteOnMainThread = new Queue<Action>();
```

Comment:
I think the Enqueue() is not thread-safe. Jon Skeet answered it for us.. :-) http://stackoverflow.com/questions/3870713/is-this-use-of-a-static-queue-thread-safe

If we would call the Enqueue() function inside a coroutine it would be okay, but that would be pretty ugly code. 

using System.Collections.Concurrent is not available in Unity. (http://answers.unity3d.com/questions/59446/is-systemcollectionsconcurrent-or-equivalent-suppo.html)

So to avoid making this too complicated on people, I'm just going to write quick object locking and access control class and link to it as a gist, will add to next PR. 

Nice catch ben!